src/Microsoft.DotNet.Interactive.Tests/StdioConnectionTests.cs

```diff
@@ -0,0 +1,158 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reactive.Disposables;
+using System.Text;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.DotNet.Interactive.Commands;
+using Microsoft.DotNet.Interactive.Connection;
+using Microsoft.DotNet.Interactive.Events;
+using Microsoft.DotNet.Interactive.Tests.Utility;
+using Microsoft.DotNet.Interactive.Utility;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.Interactive.Tests;
+
+public class StdioConnectionTests : KernelConnectionTestsBase<StdioConnectionTestConfiguration>
+{
+    public StdioConnectionTests(ITestOutputHelper output) : base(output)
+    {
+    }
+
+
+    protected override Task<IKernelConnector> CreateConnectorAsync(StdioConnectionTestConfiguration configuration)
+    {
+        var command = new List<string>{configuration.Command};
+        if (configuration.Args?.Length > 0)
+        {
+            command.AddRange(configuration.Args);
+        }
+
+        var connector = new StdIoKernelConnector(command.ToArray(), configuration.WorkingDirectory);
+
+        RegisterForDisposal(connector);
+
+        return Task.FromResult<IKernelConnector>(connector);
+    }
+
+    protected override StdioConnectionTestConfiguration CreateConnectionConfiguration()
+    {
+        var toolPath = CopyToTemp(typeof(App.Program).Assembly.Location);
+        
+        return new StdioConnectionTestConfiguration
+        {
+            Command = Dotnet.Path.FullName,
+            Args = new []
+            {
+                $"\"{toolPath.FullName}\"",
+                "stdio",
+                "--default-kernel",
+                "csharp",
+            },
+            WorkingDirectory = toolPath.Directory
+        };
+    }
+
+    private FileInfo CopyToTemp(string assemblyLocation)
```

Comment:
We have a `TemporaryDirectory` class with cleanup semantics. Maybe use that here?

Reply:
Why is this copy needed?

Reply:
using the utility, I am copying this to ensure no lock during local ncrunch / parallel test runs